src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/utils.ts

```diff
@@ -65,11 +76,14 @@ export const processSavedAddresses = (
 
 export const sortAddressesByPriority = (addresses: ProcessedUserAddress[]) => {
   return [...addresses].sort((a, b) => {
-    if (a.isDefault && a.isValid && !(b.isDefault && b.isValid)) return -1
-    if (b.isDefault && b.isValid && !(a.isDefault && a.isValid)) return 1
+    const aUsable = a.isShippable && a.isValid
+    const bUsable = b.isShippable && b.isValid
 
-    if (a.isValid && !b.isValid) return -1
-    if (!a.isValid && b.isValid) return 1
+    if (a.isDefault && aUsable && !(b.isDefault && bUsable)) return -1
+    if (b.isDefault && bUsable && !(a.isDefault && aUsable)) return 1
```

Comment:
minor: I think if a is default then b by definition is not default (and vice versa). I don't know if this is worth 'fixing' though.